pkg/client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx

```diff
@@ -265,18 +270,17 @@ export const AnalysisWizard: React.FunctionComponent<IAnalysisWizard> = ({
 
     if (createdTaskgroup) {
       const taskgroup = setTaskgroup(createdTaskgroup, data);
-      Promise.all(
```

Comment:
How were you able to get rid of the Promise.all here? Wondering because I was seeing some behavior issues resulting from promises coming back asynchronously instead of all promises resolving at once. I do not have any rules files to test this - just a note from my experience with the tag fetching in my recent PR.

Reply:
I replaced it with react query mutation, this is easier now that we're using Taskgroups, each rules file is uploaded only once for the group.

For testing, there are xml files available in "pkg/client/__mocks__/" folder 

I also realized the custom rules file upload needs to be move to the UploadFile component (triggered in the SetOptions step). 
But that can be a separate PR. Basically it would be better to upload the valid files as the user handles them.

And the other